source/dub/packagemanager.d

```diff
@@ -1138,10 +1138,11 @@ private struct Location {
 			// loaded by the user (e.g. the project and its subpackages),
 			// so don't clean it.
 			this.fromPath = null;
-			this.scanPackageFolder(this.packagePath, mgr, existing);
 		}
 		foreach (path; this.searchPath)
 			this.scanPackageFolder(path, mgr, existing);
+		if (this.packagePath !is NativePath.init)
+			this.scanPackageFolder(this.packagePath, mgr, existing);
```

Comment:
The scan order shouldn't matter, the iteration order is done in `getPackageIterator`, which IIUC is where you want to shift things.

Reply:
The problem is that both, search path packages and package path packages end up in the same array (`fromPath`) and the order within this array matters.